### DISCUSSION TITLE
Open issues

### DISCUSSION THREAD

**Discussion by edwinjosegeorge** [BerriAI/litellm#4120] [General]
Opening a discussion for bugs caused due to returning async types

connected issues are:

- https://github.com/BerriAI/litellm/issues/3862
- https://github.com/BerriAI/litellm/issues/3820
- https://github.com/BerriAI/litellm/issues/3796
- https://github.com/BerriAI/litellm/issues/3357
- https://github.com/BerriAI/litellm/issues/2332
- https://github.com/BerriAI/litellm/issues/1243

**Comment by edwinjosegeorge** [comment]
It seems the reason for failure is due to passing coroutines into fastapi, which in turn tries to convert to `dict`. However, making use of `asyncio.run(coroutine)` would result in blocking code in high traffic. Does it mean that async code may be handled like `steam=True`, as execution can be delayed?

**Comment by ghost** [comment]
these are different errors - with unrelated code paths (redis_ssl for example has nothing to do with ollama async) 

What exactly is the point of this?

**Comment by edwinjosegeorge** [comment]
It's true that they are at different code path. One thing common is that they are caused in proxy and not in SDK. Sometimes we pass coroutines into fastapi, leading to error. At times coroutines are returned from sync function. But as mentioned in https://github.com/BerriAI/litellm/pull/4102 use of `asyncio.run` would be blocking call.

The purpose of this discussion is to pool in idea in how to handle async from sync without blocking, as this error is now seen in different issues.

**Comment by ishaan-jaff** [comment]
closing this discussion - all the issues linked are unrelated and it's unclear what the point of this discussion is
